### PR TITLE
Bootstrap Maven: support for MAVEN_OPTS and properties with whitespaces

### LINKED ISSUE #11467: Command line arguments are not properly passed along when using quarkus.args
**Describe the bug**
Specifying `quarkus.args` when launching the dev mode doesn't properly pass all the arguments. It only works properly if passing a single argument that doesn't contain any space i.e. `-Dquarkus.args='--help'`. However, if you attempt to pass more complex arguments such as: `-Dquarkus.args='get abc -u foo --password foo-bar'` this will fail as the arguments get split on white spaces, resulting in an error message such as:
```
Prepare phase of the quarkus-maven-plugin threw an error: Failed to parse command line arguments [, quarkus:dev, -Dquarkus.args=get, abc, -u, foo, --password, foo-bar]: InvocationTargetException: Failed to parse Maven command line arguments: Unrecognized option: -u
```

**Expected behavior**
`quarkus.args` should be passed integrally as one arg to the maven command line parser but then should be split before Quarkus' main is run…

**Actual behavior**
See description above.

**To Reproduce**
1. Try to run any quarkus app using the dev mode with `-Dquarkus.args='get abc -u foo --password foo-bar'` for example (really anything that contains more than one arg)

**Configuration**
N/A

**Screenshots**
N/A

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64 x86_64
 - Output of `java -version`: openjdk 11.0.2 2019-01-15
OpenJDK Runtime Environment 18.9 (build 11.0.2+9)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.2+9, mixed mode)
 - GraalVM version (if different from Java): N/A
 - Quarkus version or git rev: 569fd1f41eb7a51ba7353931e729753fff305b26
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)

**Additional context**
N/A

### LINKED ISSUE #11819: Bootstrap Maven: consult MAVEN_OPTS for relevant options
Our current bootstrap code initializes the maven context from MAVEN_CMD_LINE_ARGS env var which is set by the `mvn` script and represents the command line that launched the maven build process.
However, it's not uncommon to use MAVEN_OPTS env var to set system properties (e.g. maven.repo.local, etc) and other relevant options. The value of this variable is not added to the value of MAVEN_CMD_LINE_ARGS and currently is not checked by our bootstrap code.

### PR DESCRIPTION
**Pull request by aloubyansky**
Fixes #11819
Fixes #11467

@metacosm fyi, I took a bit of a different approach here.

### CODE REVIEW

**Comment by aloubyansky** [comment]
@gsmet fyi, with this one you won't have to add `-Dmaven.repo.local=xxx` to the command line.

**Comment by metacosm** [comment]
@aloubyansky whatever you think is best

**Comment by aloubyansky** [comment]
@metacosm i think so, given that this one covers your issue plus another one.

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 1]
```diff
@@ -772,32 +772,21 @@ void prepare(final boolean triggerCompile) throws Exception {
...
-                    }
-                    i += 2;
+            for (Object o : userProps.keySet()) {
+                String name = o.toString();
+                final String value = userProps.getProperty(name);
+                buf.setLength(2);
+                buf.append(name);
+                if (value != null && !value.isEmpty()) {
+                    buf.append('=');
+                    buf.append(value);
```
Comment (gsmet) [inline devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:788]: We don't need to escape the values? They will be escaped?

**Review by gsmet — COMMENTED** [review]
Added 2 questions.

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java [hunk 2]
```diff
@@ -772,32 +772,21 @@ void prepare(final boolean triggerCompile) throws Exception {
...
-                    buf.setLength(2);
-                    i = mavenCmdLine.indexOf("-D", i);
-                    if (i < 0) {
-                        break;
-                    }
-                    i += 2;
+            for (Object o : userProps.keySet()) {
+                String name = o.toString();
+                final String value = userProps.getProperty(name);
+                buf.setLength(2);
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:784]: No. What made you question that?

File: devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java (hunk 1 above)
Comment (aloubyansky) [reply devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java:788]: It should work. This PR includes a devmode test passing a system property with a value containing whitespaces using -D.

**Review by gsmet — APPROVED** [review]

**Pull request merged**